src/version_selection.rs

```diff
@@ -0,0 +1,664 @@
+            let channel = format!(
+                "{}.{}-nightly",
+                required_version.major, required_version.minor
+            );
+            eprintln!(
+                "{} Manifest specifies Julia {} but the highest known version for {}.{} is {}. Using {} channel.",
+                console::style("Info:").cyan().bold(),
+                required,
+                required_version.major,
+                required_version.minor,
+                max_minor_version,
+                channel
+            );
+            return Ok(channel);
+        }
+    }
+
+    // Check if requested version is higher than any known version overall
+    // This handles the case where we have 1.12.x but request 1.13.0
+    let max_known_version = max_available_version(versions_db)?;
+    if let Some(max_version) = &max_known_version {
+        if &required_version > max_version {
+            // Check if a version-specific nightly channel exists (e.g., 1.13-nightly)
+            let versioned_nightly = format!(
+                "{}.{}-nightly",
+                required_version.major, required_version.minor
+            );
+
+            if versions_db
+                .available_channels
+                .contains_key(&versioned_nightly)
+            {
+                eprintln!(
+                    "{} Manifest specifies Julia {} but the highest known version is {}. Using {} channel.",
+                    console::style("Info:").cyan().bold(),
+                    required,
+                    max_version,
+                    versioned_nightly
+                );
+                return Ok(versioned_nightly);
+            }
+
+            // Fall back to main nightly channel
+            eprintln!(
+                "{} Manifest specifies Julia {} but the highest known version is {}. Using nightly channel.",
+                console::style("Info:").cyan().bold(),
+                required,
+                max_version
+            );
+            return Ok("nightly".to_string());
+        }
+    } else {
+        // No versions in database at all, use nightly
+        eprintln!(
+            "{} Manifest specifies Julia {} but no versions are known. Using nightly channel.",
+            console::style("Info:").cyan().bold(),
+            required
+        );
+        return Ok("nightly".to_string());
+    }
+
+    Err(anyhow!(
+        "Julia version `{}` requested by Project.toml/Manifest.toml is not available in the versions database.",
+        required
+    ))
+}
+
+/// Find the maximum version (including prerelease) for a given major.minor series
+pub fn max_version_for_minor(
+    versions_db: &JuliaupVersionDB,
+    major: u64,
+    minor: u64,
+) -> Result<Option<Version>> {
+    Ok(versions_db
+        .available_channels
+        .keys()
+        .filter_map(|key| parse_db_version(key).ok())
+        .filter(|version| version.major == major && version.minor == minor)
+        .max())
+}
```

Comment:
I think t his could also live in an `impl` block for JuliaupVersionDB, then you can do `versions_db.max_version_for_minor(major, minor)`. Again, doesn't really impact anything, it's just more rusty